keras/utils/data_utils.py

```diff
@@ -186,19 +186,21 @@ def get_file(fname,
 
     if download:
         print('Downloading data from', origin)
-        progbar = None
 
-        def dl_progress(count, block_size, total_size, progbar=None):
-            if progbar is None:
-                progbar = Progbar(total_size)
+        # Closures: Use a dictionary workaround To support python2,
+        # since `nonlocal` is only support in python3.
```

Comment:
@hotdogee Thanks for the very clear explanation! A [nonlocal class](http://stackoverflow.com/a/28433571/99379) may be simpler:

```python
def get_file(fname, origin):
   class progress_tracker:
        # Maintain progbar for the lifetime of download.
        # This design was chosen for Python 2.7 support.
        progbar = None

    def dl_progress(count, block_size, total_size):
        if progress_tracker.progbar is None:
            progress_tracker.progbar = Progbar(total_size)
        else:
            progress_tracker.progbar.update(count*block_size)
            
    urlretrieve(origin, fpath, dl_progress)
```

@fchollet does this approach look good to you?